src/main/java/cam72cam/immersiverailroading/render/entity/StockModel.java

```diff
@@ -116,12 +117,16 @@ private void drawStandardStock(EntityMoveableRollingStock stock) {
 		//called every tick
 		if (stock instanceof Freight) {
 			Freight freight = (Freight) stock;
+			CarFreightDefinition freightDef = (CarFreightDefinition) def;
 			int fill = freight.getPercentCargoFull();
 			List<RenderComponent> cargoLoads = def.getComponents(RenderComponentType.CARGO_FILL_X, stock.gauge);
 			if (cargoLoads != null) {
 				for (RenderComponent cargoLoad : cargoLoads) {
 					if (cargoLoad.id <= fill) {
 						drawComponent(cargoLoad);
+						if (freightDef.shouldShowCurrentLoad()) {
```

Comment:
Add a comment explaining that cargoLoads is sorted in descending order and will stop at the highest number that matches.